rajawali/src/main/java/c/org/rajawali3d/engine/EngineImpl.java

```diff
@@ -35,76 +33,90 @@
  * @author Jared Woolston (Jared.Woolston@gmail.com)
  */
 @NotThreadSafe
-public class RendererImpl implements Renderer, ISurfaceRenderer {
+public class EngineImpl implements Engine, ISurfaceRenderer {
 
-    private static final String TAG = "RendererImpl";
+    private static final String TAG = "EngineImpl";
 
-    @GuardedBy("renderables")
-    protected final List<Renderable> renderables; // List of all renderable objects this renderer is aware of.
+    // List of all RenderModels registered with the Engine
+    @GuardedBy("renderModels")
+    protected final List<RenderModel> renderModels;
+
+    // List of all RenderViews registered with the Engine
+    @GuardedBy("renderViews")
+    protected final List<RenderView> renderViews;
+
+    // List of all SurfaceCallbacks registered with the Engine
+    @GuardedBy("surfaceCallbacks")
+    protected final List<SurfaceCallback> surfaceCallbacks;
 
     @NonNull
-    protected Context context; // Context the renderer is running in
+    protected Context context; // Android context the engine is running in
 
     protected ISurface surface; // The rendering surface
 
-    protected int defaultViewportWidth;
-    protected int defaultViewportHeight; // The default width and height of the GL viewport
+    protected int surfaceWidth;
```

Comment:
Oh duh, yeah, reads could still get messed up, and volatile wouldn't cut it either (even if I had suggested it...).  Will sync!